android/app/src/main/res/values/styles.xml

```diff
@@ -2,11 +2,22 @@
 
     <!-- Base application theme. -->
     <style name="AppTheme" 
-        parent="Theme.AppCompat.Light.NoActionBar">
+        parent="Theme.AppCompat.DayNight.NoActionBar">
         <!-- This stops you seeing a white screen before the spash screen -->
         <item name="android:windowDisablePreview">true</item>
         <!-- Customize your theme here. -->
         <item name="android:textColor">#000000</item>
+        <item name="android:alertDialogTheme">@style/ThemeAlert</item>
+    </style>
+
+    <!-- https://stackoverflow.com/a/42533536 -->
+    <style name="ThemeAlert" parent="Theme.AppCompat.DayNight.Dialog.Alert">
+        <!-- Used for the buttons -->
+        <!-- <item name="colorAccent"></item> -->
+        <!-- Used for the title and text -->
+        <item name="android:textColorPrimary">#000000</item>
+        <!-- Used for the background -->
+        <item name="android:background">#ffffff</item>
     </style>
```

Comment:
yeah. i think react native made some internal changes such that some native UI elements have different interfaces based on the theme settings for the device. I'm currently on dark mode so without this change it showed up as this:

<img width="542" alt="image" src="https://user-images.githubusercontent.com/18542095/146061938-6dbc131c-14c5-47ba-a300-4524d47a8b38.png">

With this change, the alert looks like this:

<img width="551" alt="image" src="https://user-images.githubusercontent.com/18542095/146059091-547e2d94-72bd-485e-b492-94225e0993e0.png">

Found the relevant React Native issue https://github.com/facebook/react-native/issues/31345 and seems like an artifact in the `styles.xml` file is causing the text color inconsistency. Will post a screenshot with that line removed shortly

Curious if you think we should stick to letting the OS theme determine the alert style or if we should override it as i have it here?

Reply:
Here's what happens when removing `<item name="android:textColor">#000000</item>`, as suggested in the issue:

Light mode:

<img width="545" alt="image" src="https://user-images.githubusercontent.com/18542095/146062693-b05a3bd3-419b-4a29-aba5-5ac75ffa3e4a.png">

Dark mode:

<img width="533" alt="image" src="https://user-images.githubusercontent.com/18542095/146062683-60ac99b0-e303-4f66-9d32-e2a301e6d36d.png">

---

As you can see, the contrast issues are fixed but since that line is removed, the app falls back to using the OS-specified text color for native elements instead of #000000. So in my phone's case, the action buttons are teal. Basically, this line change has wider consequences and I'm not entirely sure if we're okay with it or not (maybe we are for now?). It seems like the alerts are the only UI elements that we currently use that are heavily affected by this, so maybe that's okay?

Reply:
I'm ok with either option (with or without specified text colour). We can check in the release QA if anything else is not looking right.